### ISSUE TITLE
Weird interaction when used alongside view_component_reflex

### ISSUE THREAD

**Issue by sebastianiorga** [leastbad/radiolabel#3]
# Bug
radiolabel prevents view_component_reflex controllers from initializing while the tooltip is on the screen. There doesn't seem to be the same interaction when just using regular reflex in the same application.

# Details
I added the CounterComponent example from https://github.com/joshleblanc/view_component_reflex#example to a mostly fresh rails 6.1 app and also added radiolabel in `javascript/controllers/index.js`

```
// Load all the controllers within this directory and all subdirectories.
// Controller files must be named *_controller.js.

import { Application } from "stimulus";

import { definitionsFromContext } from "stimulus/webpack-helpers";
import StimulusReflex from 'stimulus_reflex';
import consumer from '../channels/consumer';
import controller from '../controllers/application_controller';

const application = Application.start();
const context = require.context("controllers", true, /_controller\.js$/);
application.load(definitionsFromContext(context));
const debug = process.env.RAILS_ENV === 'development';
if (debug) {
  import('radiolabel').then(Radiolabel => application.register('radiolabel', Radiolabel.default));
}

StimulusReflex.initialize(application, { consumer, controller, isolate: true });
StimulusReflex.debug = debug;
```

When I click the button, it immediately updates the HTML, but only as

```
<button data-reflex="click->CounterComponent#increment" data-key="ba26f86d612c6d4fd3321d9959e11ff82ce05149661c1aa0f90969f4d7771836">Click</button>
```

Once the radiolabel div disappears the button is mutated to:

```
<button data-reflex="click->CounterComponent#increment" data-key="ba26f86d612c6d4fd3321d9959e11ff82ce05149661c1aa0f90969f4d7771836" data-controller="stimulus-reflex" data-action="click->stimulus-reflex#__perform">Click</button>
```

Disabling radiolabel makes this all instant, but to be fair non-view_component_reflex stimulus reflex elements(ie, the counter example in the stimulus reflex docs) don't have this delay. Maybe this is a problem with view_component_reflex? I'm just getting started so wasn't sure.

Thank you for this library, it makes working with SR a lot nicer!

# Dependencies
**Gems**
rails (6.1.1)
stimulus_reflex (3.4.1)
view_component (2.25.0)
view_component_reflex (3.1.5)

**Node**
"radiolabel": "1.1.0",
"stimulus": "2.0.0",
"stimulus_reflex": "3.4.1",

**Comment by leastbad** [comment]
Hello Sebastian! Thank you for filing a detailed and helpful issue.

In truth, I still haven't had the opportunity to actually try out view_component_reflex so I'm at a bit of a disadvantage. As much as I don't like to punt, the fact that you're not seeing this issue with vanilla SR does suggest that the issue is likely with VCR. That said, it goes without saying that if there's something I can easily change to make everything work, it'll be published today.

What I see when I looked at the code again for the first time in many months is that the callback which actually displays the notifications does so inside of a `setTimeout` call, which is a good thing - it allows the normal event loop to continue. This means that SR, for example, won't wait for radiolabel to finish before continuing.

This leaves me confused, then, why anything different is happening with VCR. I think that you need to file an issue on that project and/or [jump on to Discord](https://discord.gg/XveN625) and go to the `view_component` channel to ask for @joshleblanc to have a think about what might be happening.

**Comment by joshleblanc** [comment]
I'm seeing the same behaviour using a normal reflex.

I believe it's because the fade in radiolabel updates the dom, which trigger's the mutation observer in SR that sets up the declarative reflexes. Radiolabel updates the dom some ~364 times while fading out, and the declarative reflexes are trying to be setup every time that runs. 

![tSiRhw1rTa](https://user-images.githubusercontent.com/1729810/106792339-63916800-662c-11eb-870d-e21ac039ac09.gif)

You can see the mutations here, and you can see it's not a deterministic amount of time between after reflex that the data-controller gets attached. The code doing the logging is setup to mimic SR's own code.

SR:

```
  window.addEventListener('load', () => {
    setupDeclarativeReflexes()
    const observer = new MutationObserver(setupDeclarativeReflexes)
    observer.observe(document.documentElement, {
      attributes: true,
      childList: true,
      subtree: true
    })
  })
```

The logging code

```
window.addEventListener('load', () => {
    console.log("Loaded");
    const observer = new MutationObserver(() => console.log("We've mutated!"));
    observer.observe(document.documentElement, {
      attributes: true,
      childList: true,
      subtree: true
    })
})

```

So, to re-iterate. When you have a `data-reflex` tag on an element, SR's watching for DOM changes to add the `data-controller` attribute to the element. Radiolabel is updating the dom so fast that *something* is failing during SR's declarative reflex setup

@leastbad does this sound plausible?

**Comment by leastbad** [comment]
Oh, dear! That sounds about right... I hadn't even considered. This must just be *punishing* slower machines.

I need to think about what - if anything - we can do to mitigate this.

In the meantime, why is this only manifesting with VCR? Or is that a misunderstanding?

**Comment by joshleblanc** [comment]
The attached gif is just with a normal reflex

**Comment by leastbad** [comment]
> The attached gif is just with a normal reflex

Totally. I'm more alluding to the OP's suggestion that everything seemed to work fine. However, revisiting the issue, it's clear that the issue has just been hidden.

**Comment by leastbad** [comment]
Alright, there's a PR on StimulusReflex which should fix this issue. Thank you for bringing it to my attention, Sebastian. Assuming this PR is accepted, I'll help you point your Gemfile and package.json at the right place so you can use it right away.

**Comment by leastbad** [comment]
Sebastian, the branch you want to attach to (gem and npm) is https://github.com/leastbad/stimulus_reflex/tree/pause_observer

https://docs.stimulusreflex.com/hello-world/setup#running-a-branch-to-test-a-github-pull-request

This change will be included in v3.4.2 but that might not be for a few more weeks, whereas this branch will fix the issue for you tonight without you having to attach to master branch.

**Comment by sebastianiorga** [comment]
Branch works perfectly, thank you. 

Sorry about the misleading info in the OP: I only actually tested the later version of the demo, that replaced `data-reflex` with `data-controller`. Just getting started with these libraries and didn't realize it would matter.
